rpc/src/server.rs

```diff
@@ -104,3 +104,126 @@ impl RpcServer {
         }
     }
 }
+
+use crate::error::RPCError;
+use futures::{future::ok, future::Either, Async, Future, Poll};
+use jsonrpc_core::{middleware::Middleware, Call, Id, Metadata, Output, Request, Response};
+use std::sync::Arc;
+
+#[derive(Clone, Default)]
+pub struct ModuleEnableCheck(Arc<HashSet<String>>);
```

Comment:
suggest changing `HashSet<String>` to `HashMap<String, String>`  and change related module code to 
```
if self.config.chain_enable() {
  ...
} else {
  rpc_method.into_iter().each(|(method, _)| disabled_module_methods.insert("Indexer", method)) 
}
```

and we can output the detail disabled module name in the response message:

```
you need to enable `xxx` module to invoke `xxx` rpc, please modify `rpc.modules` of configuration file ckb.toml and restart the ckb node
```

Reply:
```
rpc_method.into_iter().each(|(method, _)| disabled_module_methods.insert("Indexer", method))
```
This can only record one method, maybe it should be `hashmap<method, Arc<module>>`, but this will increase the complexity of the code

Reply:
I modified comment code please check again

Reply:
Done it, please review again